src/shadertools/lib/assemble-shaders.js

```diff
@@ -120,6 +136,7 @@ ${type === FRAGMENT_SHADER ? FRAGMENT_SHADER_PROLOGUE : ''}
       inject = true;
       break;
     default:
+      checkDeprection(moduleName, coreSource);
```

Comment:
Yes, this is a good default mode. Long-term it might be nice to have a "debug" that also follows the module dependency chain and checks for deprecations in other modules using this module. 

That might be best done during module registration, rather than here. Maybe leave a TODO?